### PR TITLE
Update underscore to 1.9.2

### LINKED ISSUE #12673: Meteor-embedded underscore library contains vulnerabilities
Hi guys, is there a way for me to upgrade the meter/underscore library to a more modern version? The one that seems to be bundled with meteor 2.12 is being reported by Chrome's Lighthouse as vulnerable (attached screenshot). 

Meteor Version 2.12.
Chrome Lighthouse inspection reports a vulnerability in the underscore library.
I have it added as "underscore" in the packages file.

image

### PR DESCRIPTION
**Pull request by StorytellerCZ**
Try to fix #12673 by updating underscore to version 1.9.2, which is the latest release that uses the structure that we have and hence it is possible to update.
After looking at the changes I have also removed Meteor alterations as I don't think they are no longer valid (but will have to do some testing in regards to #594 to confirm).

### CODE REVIEW

File: packages/underscore/underscore.js [hunk 1]
```diff
@@ -65,212 +59,275 @@
...
 
-  // METEOR CHANGE: _.each({length: 5}) should be treated like an object, not an
-  // array. This looksLikeArray function is introduced by Meteor, and replaces
-  // all instances of `obj.length === +obj.length`.
-  // https://github.com/meteor/meteor/issues/594
-  // https://github.com/jashkenas/underscore/issues/770
-  var looksLikeArray = function (obj) {
-    return (obj.length === +obj.length
-            // _.isArguments not yet necessarily defined here
-            && (_isArguments(obj) || obj.constructor !== Object));
```
Comment (zodern) [inline packages/underscore/underscore.js:96]: Meteor had changed this to view objects with a length property as an object. The new code reverts it to viewing these as an array. Is the old behavior still needed?

**Comment by 07tAnYa** [comment]
I would like to contribute here. Kindly assign this to me.

**Comment by StorytellerCZ** [comment]
@07tAnYa we don't do assignments here. Submit PRs, that is all that matters.

**Comment by Grubba27** [comment]
Why does the ci does not pass?

**Comment by StorytellerCZ** [comment]
@Grubba27 TravisCI timed out. I might have possibly missed something specific that the Meteor Underscore was doing. Updating now and will check if the issue persist.

**Comment by StorytellerCZ** [comment]
Closing in favor of #12994 as that one takes more gradual approach.

**Pull request closed without merging**